docker/pypi/wmagent/bin/manage-common.sh

```diff
@@ -0,0 +1,423 @@
+}
+
+_sql_dbid_valid(){
+    # Auxiliary function to check if the build Id and hostname recorded in the database matches the $WMA_BUILD_ID
+    # :param $1: The database name to be checked (it will be ignored for Oracle)
+    echo $FUNCNAME: "Checking if the current SQL Database Id matches the WMA_BUILD_ID and hostname of the agent."
+    local wmaDBName=${1:-$wmaDBName}
+    case $AGENT_FLAVOR in
+        'oracle')
+            echo "Not implemented"
+            ;;
+        'mysql')
+            local sqlCmd="select initvalue from wma_init where initparam='wma_build_id';"
+            local dbId=$(_exec_mysql "$sqlCmd" $wmaDBName)
+            local sqlCmd="select initvalue from wma_init where initparam='hostname';"
+            local dbHostname=$(_exec_mysql "$sqlCmd" $wmaDBName)
+            if [[ $dbId == $WMA_BUILD_ID ]] && [[ $dbHostname == $HOSTNAME ]]; then
+                echo "$FUNCNAME: OK: Database recorded and current agent's init parameters match."
+                return $(true)
+            else
+                echo "$FUNCNAME: WARNING: Database recorded and current agent's init parameters do NOT match."
+                return $(false)
+            fi
+            ;;
+        *)
+            echo "$FUNCNAME: ERROR: Unknown or not set Agent Flavor"
+            return $(false)
+            ;;
+    esac
+}
+
+_sql_db_isclean(){
+    # Auxiliary function to check if a given database is empty
+    # :param $1: The database name to be checked (it will be ignored for Oracle)
+    echo "$FUNCNAME: Checking if the current SQL Database is clean and empty."
+    local wmaDBName=${1:-$wmaDBName}
+    case $AGENT_FLAVOR in
+        'oracle')
+            echo "Not implemented"
+            ;;
+        'mysql')
+            local sqlCmd="SELECT SCHEMA_NAME   FROM INFORMATION_SCHEMA.SCHEMATA  WHERE SCHEMA_NAME = '$wmaDBName'"
+            local dbExists=$(_exec_mysql "$sqlCmd")
+            [[ -n $dbExists ]] || { echo "$FUNCNAME: WARNING: Database $wmaDBName does not exist. 'manage init-agent' should be executed before starting the agent."; return $(true) ;}
+
+            local sqlCmd="SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = 'wmagent';"
+            local numTables=$(_exec_mysql "$sqlCmd" $wmaDBName)
+            [[ $numTables -eq 0 ]] || { echo "$FUNCNAME: WARNING: numTables=$numTables"; return $(false) ;}
+            ;;
+        *)
+            echo "$FUNCNAME: ERROR: Unknown or not set Agent Flavor"
+            return $(false)
+            ;;
+    esac
+}
+
+_sql_write_agentid(){
+    # Auxiliary function to write the current agent build id into the sql database
+    echo "$FUNCNAME: Preserving the current WMA_BUILD_ID and HostName at database: $wmaDBName."
+    case $AGENT_FLAVOR in
+        'oracle')
+            echo "Not implemented"
+            ;;
+        'mysql')
+            local sqlCmd=""
+
+            echo "$FUNCNAME: Creating wma_init table at database: $wmaDBName"
```

Comment:
The new table should be moved under the WMCore repository for consistency purpose. Likely under:
https://github.com/dmwm/WMCore/tree/master/src/python/WMCore/Agent/Database

Reply:
Agreed. And as you suggested in the general comment to the review, I'd prefer to deal with this in another issue and leave it as it is  currently.